### DISCUSSION TITLE
Fork of MaiPureSpaContorller + MQTT Support + Stable

### DISCUSSION THREAD

**Discussion by jgibbarduk** [UlrichMai/MaiPureSpaController#5] [Show and tell]
Hi Ulrich

Thanks for the great work on this project! 

I've taken the code and the hardware configs and run with it. I've made a fork of your code here: https://github.com/jgibbarduk/MaiPureSpaController

I added and removed a few things:

- Added: MQTT Support (logging, state and control)
- Added: Uptime monitoring (via MQTT)
- Refactored: Most of the code, to be more stable. It does not crash as much (or at all), when changing settings remotely)
- Removed: HomeKit Support (Home Assistant, which I use it with, has HomeKit built in)

Please feel free to take any code or suggest changes to my code :)

Thanks again
James.

**Comment by GitMouth** [comment]
> Hi Ulrich
> 
> Thanks for the great work on this project!
> 
> I've taken the code and the hardware configs and run with it. I've made a fork of your code here: https://github.com/jgibbarduk/MaiPureSpaController
> 
> I added and removed a few things:
> 
> * Added: MQTT Support (logging, state and control)
> * Added: Uptime monitoring (via MQTT)
> * Refactored: Most of the code, to be more stable. It does not crash as much (or at all), when changing settings remotely)
> * Removed: HomeKit Support (Home Assistant, which I use it with, has HomeKit built in)
> 
> Please feel free to take any code or suggest changes to my code :)
> 
> Thanks again James.

Hi James, 

Thank you for your great adaptation (and to Ulrich for the main code).
- Sorry if my english is not so good -
I tested your solution on a SB-H20 and an ESP D1 mini, works great.
I 'm testing MQTT functions with HA, but don't know how to get some sensors like current temperature or target temperature.
I'll be happy to get some help on this !

Bye and thank again !
